### PR TITLE
add s3 key only artifact

### PR DESCRIPTION
**Pull request by bchalk101**

### CODE REVIEW

File: src/hera/artifact.py [hunk 1]
```diff
@@ -61,3 +65,80 @@ def get_spec(self) -> IoArgoprojWorkflowV1alpha1Artifact:
...
+    This artifact is used to represent a bucket object
+
+    Attributes
+    ----------
+    name: str
+        The name of the input artifact.
+    path: str
+        The path where to store the input artifact. Note that this path is isolated from the output artifact path
+        of the previous task artifact.
+    bucket_name: str
```
Comment (flaviuvadan) [inline src/hera/artifact.py:82]:
```suggestion
    bucket: str
```

**Comment by bchalk101** [comment]
getting this error:
`HTTP response body: {"code":2,"message":"templates.artifact-passing-1fcad5d8.tasks.consumer templates.consumer.inputs.artifacts.test.from not valid in inputs"}`

**Comment by bchalk101** [comment]
yeah, so I removed the `_from` from the input artifact and that got it working. I pushed the working code, however, I feel it's a bit ugly, in the way that the knowledge of having to remove the `_from` lives inside the `Task` rather than inside the `Artifact`.

**Comment by flaviuvadan** [comment]
> knowledge of having to remove the _from lives inside the Task rather than inside the Artifact.

This concern is very reasonable. What do you think about making a `get_input_spec` or `input_from_arg_spec` on the argument objects? Wondering if each artifact/parameter can specify it's own "form" of input.

**Comment by bchalk101** [comment]
I went with `get_input_spec`, moving that to each `Artifact` as required. 

I've tested it against both minio, using the s3 artifact, and tested with the `artifact.py` example, and all is working. :)

**Pull request merged**